src/app/settings/settings.component.ts

```diff
@@ -108,8 +95,10 @@ export class SettingsComponent extends WithUnsubscribe() implements OnInit {
   }
 
   public changeLanguage(change: MatSelectChange): void {
-    this.languageService.setLanguage(change.value).subscribe();
-    this.setLoading();
+    this.isLoading = true;
```

Comment:
I believe that we need a separate loading indicator for language. and other sections